src/types/mod.rs

```diff
@@ -1992,7 +1992,12 @@ impl BlockItemSummaryDetails {
                         add_token_event_addresses(&mut addresses, events);
                         addresses.into_iter().collect()
                     }
-                }
+                };
+                // Add the optional sponsor account to the affected accounts.
+                at.sponsor
+                    .iter()
+                    .for_each(|s| affected_accounts.push(s.sponsor));
```

Comment:
Affected accounts is once again a little funny.. now that we have transaction sponsoring, is the `sender` of a `RegisterData` transaction also an "affected account"? 🤔 Maybe we should discuss this with a wider audience on slack?

Also, to me it would seem more natural to
```suggestion
                if let Some(sponsor) = at.sponsor {
                    affected_accounts.push(sponsor);
                }
```

but yours works too 🤷